rosidl_cmake/cmake/rosidl_generate_interfaces.cmake

```diff
@@ -137,22 +137,50 @@ macro(rosidl_generate_interfaces target)
   # afterwards all remaining interface files are .idl files
   list(APPEND _idl_tuples ${_idl_adapter_tuples})
 
-  # to generate action interfaces, we need to depend on "action_msgs"
+  set(_service_msgs_added FALSE)
+  set(_action_msgs_added FALSE)
   foreach(_tuple ${_interface_tuples})
     string(REGEX REPLACE ".*:([^:]*)$" "\\1" _tuple_file "${_tuple}")
     get_filename_component(_parent_dir "${_tuple_file}" DIRECTORY)
-    if("${_parent_dir}" STREQUAL "action")
+
+    if("${_parent_dir}" STREQUAL "action" AND NOT _action_msgs_added)
+      if (_service_msgs_added)
+        find_package(service_msgs QUIET)
+        if(NOT ${service_msgs_FOUND})
+          message(FATAL_ERROR
+            "Unable to generate service interface for '${_tuple_file}'. "
+            "In order to generate service interfaces you must add a depend tag "
+            "for 'service_msgs' in your package.xml.")
+        endif()
+        ament_export_dependencies(service_msgs)
+        list_append_unique(_ARG_DEPENDENCIES "service_msgs")
+        set(_service_msgs_added TRUE)
+      endif()
```

Comment:
you don't need to duplicate this if you use in L171:

```cmake
    elseif(NOT _service_msgs_added AND "${_parent_dir}" MATCHES "srv|action")
```

Reply:
I think this is not relevant with [91ea5a9](https://github.com/ros2/rosidl/pull/700/commits/91ea5a9062cecf44e8d77fc730c06ea89e922593)